src/mscorlib/src/System/Enum.cs

```diff
@@ -627,42 +614,37 @@ public static String Format(Type enumType, Object value, String format)
 
             // If the value is an Enum then we need to extract the underlying value from it
             if (valueType.IsEnum) {
-                Type valueUnderlyingType = GetUnderlyingType(valueType);
 
                 if (!valueType.IsEquivalentTo(enumType))
                     throw new ArgumentException(Environment.GetResourceString("Arg_EnumAndObjectMustBeSameType", valueType.ToString(), enumType.ToString()));
 
-                valueType = valueUnderlyingType;
-                value = ((Enum)value).GetValue();
+                if (format.Length != 1)
+                {
+                    // all acceptable format string are of length 1
+                    throw new FormatException(Environment.GetResourceString("Format_InvalidEnumFormatSpecification"));
+                }
+                return ((Enum)value).ToString(format);
             }
             // The value must be of the same type as the Underlying type of the Enum
             else if (valueType != underlyingType) {
                 throw new ArgumentException(Environment.GetResourceString("Arg_EnumFormatUnderlyingTypeAndObjectMustBeSameType", valueType.ToString(), underlyingType.ToString()));
             }
-
-            if( format.Length != 1) {
+            if (format.Length != 1)
+            {
                 // all acceptable format string are of length 1
                 throw new FormatException(Environment.GetResourceString("Format_InvalidEnumFormatSpecification"));
             }
-            
-            char formatCh = format[0];
+            if (String.Compare(format, "G", StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
I would think that the original code that just compared `char` was smaller or faster. Is there a good reason to change it to String.Compare instead?